vulture_os/daemons/reconcile.py

```diff
@@ -0,0 +1,146 @@
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Vulture OS is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Vulture OS.  If not, see http://www.gnu.org/licenses/.
+"""
+__author__ = "Olivier de Régis"
+__credits__ = ["Kevin GUILLEMOT"]
+__license__ = "GPLv3"
+__version__ = "4.0.0"
+__maintainer__ = "Vulture OS"
+__email__ = "contact@vultureproject.org"
+__doc__ = 'Job for OS Monitoring'
+
+# Django system imports
+from django.conf import settings
+
+# Extern modules imports
+import json
+from time import sleep
+from threading import Thread, Event
+
+# Logger configuration imports
+import logging
+
+logging.config.dictConfig(settings.LOG_SETTINGS)
+logger = logging.getLogger('daemon')
+
+# Redis import
+from toolkit.redis.redis_base import RedisBase
+
+# Mongo import
+from toolkit.mongodb.mongo_base import MongoBase
+
+def alert_handler(alert, m, r):
+    try:
+        a = json.loads(alert)
+    except json.JSONDecodeError as e:
+        logger.error("Alert is not JSON valid : {}".format(e))
+        return
+
+    evt_id = a.get("evt_id")
+    if evt_id is not None:
+        log = r.redis.get(evt_id)
+        if log is None:
+            logger.warning("No matching log for alert {}. Storing alert without context.".format(evt_id))
+        else:
+            try:
+                log = json.loads(log.decode())
+            except json.JSONDecodeError as e:
+                logger.error("Context is not JSON valid : {}".format(e))
+                return
+            a["context"] = log
+
+    r.redis.publish(settings.REDIS_RECONCILIED_CHANNEL, json.dumps(a))
+    m.insert("logs", "darwin_alerts", a)
+    # query = {"darwin_id": evt_id}
+    # newvalue = {"$set": {"darwin_alert_details": a, "darwin_is_alert": True}}
+    # m.update_one("logs", query, newvalue)
+    return True
+
+
+class ReconcileJob(Thread):
+
+    def __init__(self, delay):
+        super().__init__()
+        # The shutdown_flag is a threading.Event object that
+        # indicates whether the thread should be terminated.
+        self.shutdown_flag = Event()
+        self.delay = delay
+
+    def pops(self, m, r):
+        redis_list_name = settings.REDIS_LIST
+
+        logger.info("Start pops awaiting alerts.")
+        alert = r.redis.rpop(redis_list_name)
+        while (alert is not None) and (not self.shutdown_flag.is_set()):
+            alert = alert.decode()
+            alert_handler(alert, m, r)
+            alert = r.redis.rpop(redis_list_name)
+
+    def reconcile(self):
+        m = MongoBase()
+        if not m.connect():
+            return False
+        m.connect_primary()
```

Comment:
import system cluster models

cluster.current node 
is master mongo ? 
return False